geohisto/models.py

```diff
@@ -90,8 +87,71 @@ def get_current(self, depcom, valid_datetime):
         except IndexError:
             return self.latest(depcom)
 
+    def update_successors(self, town, from_town=None, to_town=None):
+        """Update references in case of a Town rename or creation."""
+        try:
+            valid_datetime = town.end_datetime + DELTA
+        except OverflowError:  # Happens on END_DATETIME + DELTA
+            valid_datetime = town.end_datetime
+        if to_town and to_town.valid_at(valid_datetime):
+            self.replace_successor(town, to_town, town.end_datetime)
+        elif from_town and from_town.valid_at(valid_datetime):
+            self.replace_successor(from_town, town)
+
+
+class Item(object):
```

Comment:
Yes but `namedtuple` is not factorizable: it not a `class` but a `class` factory and fields differs between `Town` and `Intercommunality`